Use full prototype for pt_solaris_sendfilev_fptr to avoid GCC 15 error.

Should be sent to upstream.

--- nspr-4.35/nspr/pr/src/pthreads/ptio.c
+++ nspr-4.35/nspr/pr/src/pthreads/ptio.c
@@ -79,7 +79,7 @@ #define SFV_FD_SELF (-2)
 /*
  * extern ssize_t sendfilev(int, const struct sendfilevec *, int, size_t *);
  */
-static ssize_t (*pt_solaris_sendfilev_fptr)() = NULL;
+static ssize_t (*pt_solaris_sendfilev_fptr)(int, const struct sendfilevec *, int, size_t *) = NULL;
 
 #define SOLARIS_SENDFILEV(a, b, c, d) \
         (*pt_solaris_sendfilev_fptr)((a), (b), (c), (d))
@@ -2722,7 +2722,7 @@ PR_LOG(_pr_io_lm, PR_LOG_DEBUG,
                ("dlopen(0) returns %p", handle));
         close_it = PR_TRUE;
     }
-    pt_solaris_sendfilev_fptr = (ssize_t (*)()) dlsym(handle, "sendfilev");
+    pt_solaris_sendfilev_fptr = (ssize_t (*)(int, const struct sendfilevec *, int, size_t *)) dlsym(handle, "sendfilev");
     PR_LOG(_pr_io_lm, PR_LOG_DEBUG,
            ("dlsym(sendfilev) returns %p", pt_solaris_sendfilev_fptr));
 
